Docs: Add public roadmap

shilman opened the pull request:
Add public roadmap to the documentation

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

shilman committed:
Docs: Add public roadmap

kylegach commented on docs/toc.js:
```diff
@@ -846,6 +846,11 @@ module.exports = {
           type: 'link',
         },
+        {
+          pathSegment: 'public-roadmap',
```
Maybe just `roadmap` (with the content filename adjusted to match)?

I think the "public" is redundant with the fact that it's published on our docs.

kylegach committed:
Update URL to `/contribute/roadmap`

- Update heading structure
- Clarified shaping process
- Typos and other small tweaks

kylegach committed:
Typo

kylegach reviewed (approved):
Looks great!
